api_test.go

```diff
@@ -71,6 +72,203 @@ func TestCreateProxyBlankUpstream(t *testing.T) {
+		} else if _, ok := proxies["test"]; !ok {
+			t.Fatalf("Wrong proxy name returned")
+		}
+
+		for _, p := range proxies {
+			AssertProxyUp(t, p.Listen, true)
+		}
+	})
+}
+
+func TestPopulateExistingProxy(t *testing.T) {
+	WithServer(t, func(addr string) {
+		testProxy, err := client.CreateProxy("one", "localhost:7070", "localhost:7171")
+		if err != nil {
+			t.Fatal("Unable to create proxy:", err)
+		}
+		_, err = client.CreateProxy("two", "localhost:7373", "localhost:7474")
+		if err != nil {
+			t.Fatal("Unable to create proxy:", err)
+		}
+
+		// Create a toxic so we can make sure the proxy wasn't replaced
+		_, err = testProxy.AddToxic("", "latency", "downstream", 1, nil)
+		if err != nil {
+			t.Fatal("Unable to create toxic:", err)
+		}
+
+		testProxies, err := client.Populate([]tclient.Proxy{
+			{
+				Name:     "one",
+				Listen:   "127.0.0.1:7070", // TODO(xthexder): Will replace existing proxy if not resolved ip...
+				Upstream: "localhost:7171",
+				Enabled:  true,
+			},
+			{
+				Name:     "two",
+				Listen:   "127.0.0.1:7575",
+				Upstream: "localhost:7676",
+				Enabled:  true,
+			},
+		})
+
+		if err != nil {
+			t.Fatal("Unable to populate:", err)
+		} else if len(testProxies) != 2 {
+			t.Fatalf("Wrong number of proxies returned: %d != 2", len(testProxies))
+		} else if testProxies[0].Name != "one" || testProxies[1].Name != "two" {
+			t.Fatalf("Wrong proxy names returned: %s, %s", testProxies[0].Name, testProxies[1].Name)
+		} else if testProxies[0].Listen != "127.0.0.1:7070" || testProxies[1].Listen != "127.0.0.1:7575" {
+			t.Fatalf("Wrong proxy listen addresses returned: %s, %s", testProxies[0].Listen, testProxies[1].Listen)
+		}
+
+		toxics, err := testProxy.Toxics()
+		if err != nil {
+			t.Fatal("Unable to get toxics:", err)
+		}
+		if len(toxics) != 1 || toxics[0].Type != "latency" {
+			t.Fatalf("Populate did not preseve existing proxy. (%d toxics)", len(toxics))
+		}
+
+		for _, p := range testProxies {
+			AssertProxyUp(t, p.Listen, true)
+		}
+	})
+}
+
+func TestPopulateWithBadName(t *testing.T) {
+	WithServer(t, func(addr string) {
+		testProxies, err := client.Populate([]tclient.Proxy{
+			{
+				Name:     "one",
+				Listen:   "localhost:7070",
+				Upstream: "localhost:7171",
+				Enabled:  true,
+			},
+			{
+				Name:    "",
+				Listen:  "",
+				Enabled: true,
+			},
+		})
+
+		if err == nil {
+			t.Fatal("Expected Populate to fail.")
+		} else if err.Error() != "Populate: HTTP 400: missing required field: name" {
+			t.Fatal("Expected different error during populate:", err)
+		} else if len(testProxies) != 0 {
+			t.Fatalf("Wrong number of proxies returned: %d != 0", len(testProxies))
+		}
+
+		proxies, err := client.Proxies()
+		if err != nil {
+			t.Fatal(err)
+		} else if len(proxies) != 0 {
+			t.Fatalf("Expected no proxies to be created: %d != 0", len(proxies))
+		}
+	})
+}
+
+func TestPopulateProxyWithBadDataShouldReturnError(t *testing.T) {
```

Comment:
Probably a good idea, since just enabling it all the time currently passes tests.